File: backend/app/schemas/attachment.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime
from uuid import UUID


class AttachmentUploadResponse(BaseModel):
    """附件上传响应模型"""
    attachment_ids: List[UUID] = Field(..., description="上传的附件ID列表")
    uploaded_count: int = Field(..., description="成功上传的文件数量")


class AttachmentInfo(BaseModel):
    """附件信息模型"""
    id: UUID = Field(..., description="附件ID")
    evaluation_id: UUID = Field(..., description="自评表ID")
    indicator: str = Field(..., description="考核指标")
    file_name: str = Field(..., description="文件名")
    file_size: int = Field(..., description="文件大小（字节）")
    file_type: str = Field(..., description="文件类型")
    storage_path: str = Field(..., description="存储路径")
    classified_by: str = Field(..., description="分类方式: user 或 ai")
    uploaded_at: datetime = Field(..., description="上传时间")
    is_archived: bool = Field(..., description="是否已归档")
    archived_at: datetime = Field(..., description="归档时间")

    class Config:
        from_attributes = True


class AttachmentWithRelations(AttachmentInfo):
    """附件信息模型（包含关联信息）- 需求 18.2, 18.3"""
    teaching_office_id: Optional[UUID] = Field(None, description="关联的教研室ID")
    teaching_office_name: Optional[str] = Field(None, description="关联的教研室名称")
    evaluation_year: Optional[int] = Field(None, description="考核年度")


class AttachmentClassificationUpdate(BaseModel):
    """附件分类更新请求模型"""
    indicator: str = Field(..., description="新的考核指标分类")


class AttachmentClassificationResponse(BaseModel):
    """附件分类更新响应模型"""
    id: UUID = Field(..., description="附件ID")
    indicator: str = Field(..., description="更新后的考核指标")
    message: str = Field(..., description="操作结果消息")

    class Config:
        from_attributes = True


class AttachmentQueryParams(BaseModel):
    """附件查询参数模型 - 需求 18.5"""
    teaching_office_id: Optional[UUID] = Field(None, description="按教研室ID筛选")
    indicator: Optional[str] = Field(None, description="按考核指标筛选")
    evaluation_year: Optional[int] = Field(None, description="按考核年度筛选")
    is_archived: Optional[bool] = Field(None, description="按归档状态筛选")

File: frontend/src/types/attachment.ts
// Attachment types
export interface Attachment {
  id?: string;
  evaluation_id?: string;
  evaluationId?: string;
  indicator: string; // 考核指标
  file_name?: string;
  fileName?: string;
  file_size?: number;
  fileSize?: number;
  file_type?: string;
  fileType?: string;
  storage_path?: string;
  storagePath?: string;
  classified_by?: 'user' | 'ai';
  classifiedBy?: 'user' | 'ai';
  uploaded_at?: string;
  uploadedAt?: string;
  is_archived?: boolean;
  isArchived?: boolean;
  archived_at?: string;
  archivedAt?: string;
}

// Attachment with relations (includes teaching office and evaluation info)
export interface AttachmentWithRelations extends Attachment {
  teaching_office_id?: string;
  teaching_office_name?: string;
  evaluation_year?: number;
}

export interface AttachmentUploadData {
  evaluationId: string;
  indicator: string;
  files: File[];
}

export interface UploadProgress {
  fileName: string;
  percentage: number;
  status: 'uploading' | 'success' | 'error';
}

// 考核指标类型
export type IndicatorType = 'teaching_reform_projects' | 'honorary_awards' | 'course_construction' | 'teaching_quality' | 'student_guidance' | 'scientific_research' | 'team_building';

// 考核指标配置
export interface IndicatorConfig {
  key: IndicatorType;
  label: string;
  description: string;
  fileTypes: string[]; // 支持的文件类型
  category: 'certificate' | 'project'; // 证书类或项目类
}

// 预定义的考核指标
export const INDICATORS: IndicatorConfig[] = [
  {
    key: 'teaching_reform_projects',
    label: '教学改革项目',
    description: '教学改革项目相关证书、立项文件等',
    fileTypes: ['pdf', 'doc', 'docx', 'jpg', 'jpeg', 'png'],
    category: 'project'
  },
  {
    key: 'honorary_awards',
    label: '荣誉表彰',
    description: '荣誉证书、表彰文件等',
    fileTypes: ['pdf', 'doc', 'docx', 'jpg', 'jpeg', 'png'],
    category: 'certificate'
  },
  {
    key: 'course_construction',
    label: '课程建设',
    description: '课程建设相关材料',
    fileTypes: ['pdf', 'doc', 'docx', 'ppt', 'pptx'],
    category: 'project'
  },
  {
    key: 'teaching_quality',
    label: '教学质量',
    description: '教学质量评估材料',
    fileTypes: ['pdf', 'doc', 'docx', 'xls', 'xlsx'],
    category: 'project'
  },
  {
    key: 'student_guidance',
    label: '学生指导',
    description: '学生指导相关材料',
    fileTypes: ['pdf', 'doc', 'docx'],
    category: 'project'
  },
  {
    key: 'scientific_research',
    label: '科研工作',
    description: '科研成果、论文等材料',
    fileTypes: ['pdf', 'doc', 'docx'],
    category: 'project'
  },
  {
    key: 'team_building',
    label: '团队建设',
    description: '团队建设相关材料',
    fileTypes: ['pdf', 'doc', 'docx', 'jpg', 'jpeg', 'png'],
    category: 'project'
  }
];
